File: TextosJapones/src/main/java/org/jisho/textosJapones/model/entities/novelextractor/NovelTabela.java
package org.jisho.textosJapones.model.entities.novelextractor;

import org.jisho.textosJapones.model.entities.Novel;

import java.util.ArrayList;
import java.util.List;

public class NovelTabela extends Novel {

    private List<NovelVolume> volumes;

    public List<NovelVolume> getVolumes() {
        return volumes;
    }

    public void setVolumes(List<NovelVolume> volumes) {
        this.volumes = volumes;
    }

    public void addVolume(NovelVolume volume) {
        this.volumes.add(volume);
    }

    public NovelTabela() {
        this.base = "";
        this.volumes = new ArrayList<>();
    }

    public NovelTabela(String base, List<NovelVolume> volumes) {
        this.base = base;
        this.volumes = volumes;
    }

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/model/entities/novelextractor/NovelTabela.kt
package org.jisho.textosJapones.model.entities.novelextractor

import br.com.fenix.processatexto.model.entities.Novel
import br.com.fenix.processatexto.model.entities.novelextractor.NovelVolume

data class NovelTabela(
    val volumes: MutableList<NovelVolume> = mutableListOf()
) : Novel() {

    constructor(
        tabela: String,
        volumes: MutableList<NovelVolume>
    ) : this(volumes) {
        this.base = tabela
    }

    fun addVolume(volume: NovelVolume) = volumes.add(volume)
}